File: top-k-frequent-elements/top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> countMap = new HashMap();
        Set<Integer> uniqueNums = new HashSet();
        PriorityQueue<Integer> queue = new PriorityQueue<>(
            (a,b) ->
            {
                return
                    countMap.getOrDefault(b, 0) - countMap.getOrDefault(a, 0); 
            });
        
        for(int item : nums)
        {
            int count = countMap.getOrDefault(item, 0); 
            
            countMap.put(item, count+1);
            uniqueNums.add(item);
        }
        
        for(int item : uniqueNums)
        {
            queue.add(item);
        }
        
        int[] arr = new int[k];
        int i = 0;
        
        while(i < k && !queue.isEmpty())
        {
            arr[i++] = queue.poll();
        }
        
        return arr;
        
        
    }
}

File: top-k-frequent-elements/top-k-frequent-elements.py

class Solution(object):   
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        
        hashMap = {}
        
        for item in nums:
            if item in hashMap:
                hashMap[item] +=1
            else:
                hashMap[item] = 1
                
        heap = []
        
        for key in hashMap:
            heapq.heappush(heap, (-hashMap[key], key))
            
        result = []
        
        for _ in range(k):
            pop = heapq.heappop(heap)
            result.append(pop[1])
        
        return result;
        